klient/internal/testutil/setup.go

```diff
@@ -143,3 +158,28 @@ func waitForControlPlane(c kubernetes.Interface) error {
 		})
 	return err
 }
+
+// Ensures all pods for clusterAPI are running.
+func waitForClusterAPI(c kubernetes.Interface) error {
+	log.Info("Waiting for clusterAPI pods to be initialized...")
+	err := wait.Poll(5*time.Second, time.Minute*2,
+		func() (bool, error) {
+			pods, err := c.CoreV1().Pods("capi-system").List(context.TODO(), metav1.ListOptions{})
```

Comment:
The only pod is the capi-controller and if that isnt running then the webhook throws an error when the test tries to create a cluster object.